### PR TITLE
RTKQ: Add typing to RTKQ hooks in apiSlice

### PR DESCRIPTION
**Pull request by lucasgarfield**
This commit adds typing to the RTK Query hooks. The hooks have been refactored to be simpler and all custom query functions have been removed.

### CODE REVIEW

**Comment by lucasgarfield** [comment]
Note that this PR is rebased on #1180.

File: src/Components/CreateImageWizard/formComponents/AWSSourcesSelect.js [hunk 1]
```diff
@@ -107,6 +132,17 @@ export const AWSSourcesSelect = ({
...
           </Alert>
         )}
+        {!isError && isErrorDetails && (
+          <Alert
+            variant={'danger'}
+            isPlain
+            isInline
+            title={'AWS details unavailable'}
+          >
+            Could not fetch Account id from AWS for given Source. Check Sources
```
Comment (regexowl) [inline src/Components/CreateImageWizard/formComponents/AWSSourcesSelect.js:142]: Can we change "Account id" to "account ID" so it's the same as in "Manually enter an account ID" radio? Or capitalize ID and unify it with "Associated Account ID". 

(we should probably go through the capitalization as a whole at some point, it's not super consistent)

**Comment by regexowl** [comment]
Just one nitpick and one question, otherwise I'd say it's looking good :+1:

File: src/test/fixtures/repositories.js [hunk 2]
```diff
@@ -1,15 +1,28 @@
-export const mockRepositoryResults = (args) => {
-  const data = generateData(args);
-  const meta = generateMeta(args.limit, data.length);
-  const links = generateLinks(args.limit);
-  return {
+/* TODO
```
[the remark this answers is not in the document]
Comment (lucasgarfield) [reply src/test/fixtures/repositories.js:1]: No, it is not... :smiling_face_with_tear:

**Review by regexowl — APPROVED** [review]
Great job! Thanks! :sparkles:

**Pull request merged**